### PR TITLE
fix issue 16090 - popFront generates out-of-bounds array index on cor…

### PR DESCRIPTION
**Pull request by aG0aep6G**
…rupted utf-8 strings

https://issues.dlang.org/show_bug.cgi?id=16090

### CODE REVIEW

**Comment by DmitryOlshansky** [comment]
LGTM

File: std/range/primitives.d [hunk 1]
```diff
@@ -2104,13 +2106,14 @@ if (isNarrowString!(C[]))
                  //Invalid UTF-8
                  msbs = 1;
              }
-             str = str[msbs .. $];
+             str = str.ptr[min(msbs, str.length) .. str.length];
```
Comment (JackStouffer) [inline std/range/primitives.d]: Is the access to `.ptr` really nessesary here?

Comment (aG0aep6G) [inline std/range/primitives.d]: It's not necessary, but the implementation seems to care about avoiding bounds checking. Compare [lines 2097 and 2098](https://github.com/dlang/phobos/pull/4387/files/e1af1b0b51ea9f29d4ff8076d73c03ba10bfc73c#diff-9f63c74383984a09f5bf578493892e27R2097) above.

**Comment by schveiguy** [comment]
I agree let's not corrupt memory. Should an exception be thrown though instead of silently working, as `front` does?

**Comment by aG0aep6G** [comment]
> Should an exception be thrown though instead of silently working, as front does?

I went with the silent behavior, because it doesn't throw on invalid bytes either (cf. "invalid UTF-8" comment above the changes).

**Comment by schveiguy** [comment]
> I went with the silent behavior, because it doesn't throw on invalid bytes either

Yep, good point.

**Comment by schveiguy** [comment]
Auto-merge toggled on

File: std/range/primitives.d (hunk 1 above)
Comment (andralex) [inline std/range/primitives.d]: yah this is legit

**Comment by andralex** [comment]
thx!

**Pull request merged**